sdk/ai/ai-projects/review/ai-projects.api.md

```diff
@@ -0,0 +1,1988 @@
+
+```ts
+
+import { AbortSignalLike } from '@azure/abort-controller';
+import { ClientOptions } from '@azure-rest/core-client';
+import { Paged } from '@azure/core-paging';
+import { RequestParameters } from '@azure-rest/core-client';
+import { StreamableMethod } from '@azure-rest/core-client';
+import { TokenCredential } from '@azure/core-auth';
+
+// @public
+export interface AgentDeletionStatusOutput {
+    deleted: boolean;
+    id: string;
+    object: "assistant.deleted";
+}
+
+// @public
+export interface AgentEventMessage {
+    data: AgentThreadOutput | ThreadRunOutput | RunStepOutput | ThreadMessageOutput | MessageDeltaChunk | RunStepDeltaChunk | string;
+    event: AgentStreamEventType | string;
+}
+
+// @public
+export interface AgentEventMessageStream extends AsyncDisposable, AsyncIterable<AgentEventMessage> {
+}
+
+// @public
+export interface AgentOutput {
+    created_at: number;
+    description: string | null;
+    id: string;
+    instructions: string | null;
+    metadata: Record<string, string> | null;
+    model: string;
+    name: string | null;
+    object: "assistant";
+    response_format?: AgentsApiResponseFormatOptionOutput | null;
+    temperature: number | null;
+    tool_resources: ToolResourcesOutput | null;
+    tools: Array<ToolDefinitionOutput>;
+    top_p: number | null;
+}
+
+// @public
+export type AgentRunResponse = PromiseLike<ThreadRunOutput> & {
+    stream: () => Promise<AgentEventMessageStream>;
+};
+
+// @public
+export interface AgentsApiResponseFormat {
+    type?: ApiResponseFormat;
+}
+
+// @public
+export type AgentsApiResponseFormatMode = string;
+
+// @public
+export type AgentsApiResponseFormatModeOutput = string;
+
+// @public
+export type AgentsApiResponseFormatOption = string | AgentsApiResponseFormatMode | AgentsApiResponseFormat;
+
+// @public
+export type AgentsApiResponseFormatOptionOutput = string | AgentsApiResponseFormatModeOutput | AgentsApiResponseFormatOutput;
+
+// @public
+export interface AgentsApiResponseFormatOutput {
+    type?: ApiResponseFormatOutput;
+}
+
+// @public
+export type AgentsApiToolChoiceOption = string | AgentsApiToolChoiceOptionMode | AgentsNamedToolChoice;
+
+// @public
+export type AgentsApiToolChoiceOptionMode = string;
+
+// @public
+export type AgentsApiToolChoiceOptionModeOutput = string;
+
+// @public
+export type AgentsApiToolChoiceOptionOutput = string | AgentsApiToolChoiceOptionModeOutput | AgentsNamedToolChoiceOutput;
+
+// @public
+export interface AgentsNamedToolChoice {
+    function?: FunctionName;
+    type: AgentsNamedToolChoiceType;
+}
+
+// @public
+export interface AgentsNamedToolChoiceOutput {
+    function?: FunctionNameOutput;
+    type: AgentsNamedToolChoiceTypeOutput;
+}
+
+// @public
+export type AgentsNamedToolChoiceType = string;
+
+// @public
+export type AgentsNamedToolChoiceTypeOutput = string;
+
+// @public (undocumented)
+export interface AgentsOperations {
```

Comment:
Can you help me understand whether these are rest-level clients that are all autogenerated or all were hand authored? That'll help me review

Reply:
These are hand crafted, this was an aligned to use this approach as there were issue with DPG